### ISSUE TITLE
unfold _ at -1 doesn't work when there is only one occurrence

### ISSUE THREAD

**Issue by bcip** [coq/coq#8261]
#### Version

The Coq Proof Assistant, version 8.8.1 (July 2018)
compiled on Jul 3 2018 10:30:00 with OCaml 4.06.1

#### Operating system
Windows 10

#### Description of the problem

```
Goal forall (m:nat), m = 0.
  intros.
  set (n := 0).
  unfold n at -1.
(*
1 subgoal
m : nat
n := 0 : nat
______________________________________(1/1)
m = n
*)
```

**Comment by bcip** [comment]
I realize `unfold n at -1` means to unfold `n` everywhere else than the first occurrence? Then how can I unfold only the last occurrence?

**Comment by Zimmi48** [comment]
`at -1` is not even documented for `unfold` BTW (but it is for `pattern`): https://coq.inria.fr/refman/proof-engine/tactics.html#coq:tacn.unfold

Anyways, it seems then that your issue is an enhancement request for a way to count occurrences from right to left.

**Comment by bcip** [comment]
@Zimmi48 Thanks for noticing. Actually, behavior of `at -1` is described [here](https://coq.inria.fr/refman/proof-engine/tactics.html#occurrences-sets-and-occurrences-clauses).
I find what I want to do can be implemented by posing a variable to protect the part that I don't what to unfold.
But providing a way to count occurrences reversely would be a useful feature.

**Comment by Zimmi48** [comment]
Right, thanks for pointing this out. But since we define the grammar of occurrences, we should refer to it in the doc of `unfold` (and `pattern`), instead of providing an approximate documentation there.
